tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/asynchronous/AsyncApplicationScopeClient.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package org.eclipse.microprofile.fault.tolerance.tck.asynchronous;
+
+import java.util.concurrent.Future;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.fault.tolerance.tck.util.Connection;
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+
+@ApplicationScoped
+public class AsyncApplicationScopeClient {
+
+    @Inject
+    private AsyncRequestScopeClient requestScopeClient;
+
+    @Asynchronous
+    public Future<Connection> service() {
+        // AsyncRequestScopeClient is a request scope bean
+        return CompletableFutureHelper.toCompletableFuture(requestScopeClient.sampleMethod());
```

Comment:
I think the return type of `AsyncRequestScopeClient.sampleMethod` could also be `Future<Connection>`, because `CompletableFuture` extends `Future`. This method would then just forward the call.